panacea/web/static/js/app.js

```diff
@@ -25,64 +25,80 @@ async function submitFile() {
       headers: new Headers({authorization: apiAccessToken})
     });
 
-    if (drugsResponse.ok) {
-      const {drugs} = await drugsResponse.json();
-      const labels = drugs.map(x => x.label);
-
-      if (labels.length > 0) {
-        displayDrugs(labels);
-
-        const urisResponse = await fetch('/api/uris', {
-          method: 'POST',
-          body: JSON.stringify({labels: labels}),
-          credentials: 'same-origin',
-          headers: defaultHeaders
-        });
-
-        if (urisResponse.ok) {
-          const {uris: {found, not_found: unidentifiedDrugs}} = await urisResponse.json();
-          const uris = found.map(x => x.uri);
-
-          if (unidentifiedDrugs.length > 0) {
-            displayUnidentifiedDrugs(unidentifiedDrugs);
-          }
-
-          if (uris.length >= 2) {
-            const ddisResponse = await fetch('/api/ddis', {
-              method: 'POST',
-              body: JSON.stringify({drugs: uris}),
-              credentials: 'same-origin',
-              headers: defaultHeaders
-            });
-
-            if (ddisResponse.ok) {
-              const {ddis} = await ddisResponse.json();
-              displayDdis(ddis);
-            } else {
-              const {error} = await ddisResponse.json();
-              displayError(error);
-            }
-          } else {
-            console.log("by definition ddis require more than one drug")
-          }
-        } else {
-          const {error} = await urisResponse.json();
-          displayError(error);
-        }
-      } else {
-        displayError({title: "Pathway error", detail: "No drugs found"});
-      }
-    } else {
-      const {error} = await drugsResponse.json();
-      displayError(error);
-    }
+    handleDrugsResponse(drugsResponse);
+
     this.reset();
   } catch (err) {
     console.log(err);
   }
   restoreFileForm();
 }
 
+async function handleResponse(response, handle) {
+  if (response.ok) {
+    const payload = await response.json();
+    handle(payload);
+  } else {
+    const {error} = await response.json();
+    displayError(error);
+  }
+}
+
+async function handleDrugsResponse(drugsResponse) {
+  handleResponse(drugsResponse, async function (payload) {
+    const {drugs} = payload;
```

Comment:
https://github.com/tom-and-the-toothfairies/pathways/pull/117/files#diff-af326ae966b05ea64b817eac7636661eR72 for example, i didn't know how best to generalise the destructuring